tsl/src/compression/deltadelta.c

```diff
@@ -568,6 +569,44 @@ delta_delta_decompression_iterator_try_next_forward(DecompressionIterator *iter)
 								 iter->element_type);
 }
 
+#define ELEMENT_TYPE uint64
+#include "simple8b_rle_decompress_all.h"
+#undef ELEMENT_TYPE
+
+/* Functions for bulk decompression. */
+#define ELEMENT_TYPE uint16
+#include "deltadelta_impl.c"
+#undef ELEMENT_TYPE
+
+#define ELEMENT_TYPE uint32
+#include "deltadelta_impl.c"
+#undef ELEMENT_TYPE
+
+#define ELEMENT_TYPE uint64
+#include "deltadelta_impl.c"
+#undef ELEMENT_TYPE
+
+ArrowArray *
+delta_delta_decompress_all_forward_direction(Datum compressed_data, Oid element_type)
+{
+	switch (element_type)
+	{
+		case INT8OID:
+		case TIMESTAMPOID:
+		case TIMESTAMPTZOID:
+			return delta_delta_decompress_all_uint64(compressed_data);
+		case INT4OID:
+		case DATEOID:
+			return delta_delta_decompress_all_uint32(compressed_data);
+		case INT2OID:
+			return delta_delta_decompress_all_uint16(compressed_data);
+		default:
+			elog(ERROR, "type oid %d is not supported for deltadelta decompression", element_type);
+			return NULL;
```

Comment:
In other functions (e.g.,  `ts_internal_to_interval_value`) we use `pg_unreachable();` instead of returning `NULL` from the default case.

Reply:
yes we should use pg_unreachable, this helps certain compilers figure out that the function does not return and prevents false positive compiler warnings